packages/react-dom/src/events/__tests__/DOMEventListenerSystem-test-internal.js

```diff
@@ -0,0 +1,865 @@
+    }
+
+    ReactDOM.render(<Test />, container);
+
+    let buttonElement = buttonRef.current;
+    dispatchClickEvent(buttonElement);
+    expect(rootListerner1).toHaveBeenCalledTimes(1);
+    expect(rootListerner2).toHaveBeenCalledTimes(1);
+    expect(rootListerner3).toHaveBeenCalledTimes(1);
+    expect(rootListerner4).toHaveBeenCalledTimes(1);
+  });
+
+  it('should correctly handle updating listeners #1', () => {
+    const buttonRef = React.createRef();
+    const targetListerner1 = jest.fn();
+    const targetListerner2 = jest.fn();
+
+    function Test({toggle}) {
+      const listerner1 = ReactDOM.unstable_createListener(
+        'click',
+        targetListerner1,
+      );
+      const listerner2 = ReactDOM.unstable_createListener(
+        'click',
+        targetListerner2,
+      );
+
+      return (
+        <button
+          listeners={toggle ? [listerner1, listerner2] : []}
+          ref={buttonRef}>
+          Click me!
+        </button>
+      );
+    }
+
+    ReactDOM.render(<Test toggle={true} />, container);
+
+    let buttonElement = buttonRef.current;
+    dispatchClickEvent(buttonElement);
+    expect(targetListerner1).toHaveBeenCalledTimes(1);
+    expect(targetListerner2).toHaveBeenCalledTimes(1);
+
+    ReactDOM.render(<Test toggle={false} />, container);
+
+    buttonElement = buttonRef.current;
+    dispatchClickEvent(buttonElement);
+    expect(targetListerner1).toHaveBeenCalledTimes(1);
+    expect(targetListerner2).toHaveBeenCalledTimes(1);
+  });
+
+  it('should correctly handle updating listeners #2', () => {
+    const buttonRef = React.createRef();
+    const targetListerner1 = jest.fn();
+    const targetListerner2 = jest.fn();
+
+    function Test({toggle}) {
+      const listerner1 = ReactDOM.unstable_createListener(
+        'click',
+        targetListerner1,
+      );
+      const listerner2 = ReactDOM.unstable_createListener(
+        'click',
+        targetListerner2,
+      );
+
+      return (
+        <button
+          listeners={toggle ? [listerner1] : [listerner1, listerner2]}
+          ref={buttonRef}>
+          Click me!
+        </button>
+      );
+    }
+
+    ReactDOM.render(<Test toggle={true} />, container);
+
+    let buttonElement = buttonRef.current;
+    dispatchClickEvent(buttonElement);
+    expect(targetListerner1).toHaveBeenCalledTimes(1);
+    expect(targetListerner2).toHaveBeenCalledTimes(0);
+
+    ReactDOM.render(<Test toggle={false} />, container);
+
+    buttonElement = buttonRef.current;
+    dispatchClickEvent(buttonElement);
+    expect(targetListerner1).toHaveBeenCalledTimes(2);
+    expect(targetListerner2).toHaveBeenCalledTimes(1);
+
+    ReactDOM.render(<Test toggle={true} />, container);
+
+    buttonElement = buttonRef.current;
+    dispatchClickEvent(buttonElement);
+    expect(targetListerner1).toHaveBeenCalledTimes(3);
+    expect(targetListerner2).toHaveBeenCalledTimes(1);
+  });
+
+  it('should correctly handle updating listeners #3', () => {
```

Comment:
All these test descriptions could benefit from improvement. If one of these tests fail, how can we help someone on the team quickly understand what this is testing for?